lib/perl/Genome/Process.pm

```diff
@@ -710,12 +710,22 @@ sub _compare_output_directories {
                 my $rel_path = File::Spec->abs2rel($file, $output_dir);
                 $diffs{$rel_path} = sprintf($template, $type, $rel_path, $other_process->id);
             } else {
-                if (-d $target && -d $other_target) {
-                    my %additional_diffs = $self->_compare_output_directories($target, $other_target, $other_process);
-                    %diffs = (%diffs, %additional_diffs);
-                }
-                elsif (-f $target && -f $other_target && !compare($target, $other_target)) {
-                    #Files are in fact the same - do nothing
+                if (-l $file || -l $other_file) {
+                    if (-d $target && -d $other_target) {
+                        my %additional_diffs = $self->_compare_output_directories($target, $other_target, $other_process);
+                        %diffs = (%diffs, %additional_diffs);
+                    }
+                    if (-f $target && -f $other_target) {
+                        if (compare($target, $other_target)) {
+                            $diffs{File::Spec->abs2rel($file, $output_dir)} = sprintf(
+                                'symlinks are not the same (diff -u %s %s) (diff -u %s %s)',
+                                $file, $other_file, $target, $other_target
```

Comment:
Not necessarily. I guess if we assume that it will always be called by _compare_output_files, they will be. I originally was only printing the targets but I changed that after @nnutter's comment on PR #391.